### PR TITLE
docs(theme): upgrade mkdocs material to 8.x

### PR DESCRIPTION
**Pull request by dreamorosi**
## Description of your changes

This PR updates MKDocs (the documentation engine we use) and adds the necessary extension to add annotations on code (no annotation added at this time). The PR mirrors [this commit from the Python's repo](https://github.com/awslabs/aws-lambda-powertools-python/commit/207b3df7c249fcfb276961c1ebb99844d831125f).

### How to verify this change

Checkout repo & build docs locally.

### Related issues, RFCs

N/A

### PR status

***Is this ready for review?:*** NO  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] My changes generate *no new warnings*
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

N/A

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by saragerion — APPROVED** [review]
LGTM already

File: .gitignore [hunk 1]
```diff
@@ -36,6 +36,7 @@ venv
 
 # Static documentation site generated by Mkdocs
 site
+!docs/overrides/*.html
```
Comment (ijemmy) [inline .gitignore:39]: Why do we need an override to be included in git. Is it being ignored somewhere else?

Comment (ijemmy) [reply .gitignore:39]: If it really needs to be in this file, a comment would be useful for future contributors.

**Review by ijemmy — COMMENTED** [review]
I have one small comment. Please ping me in Slack after it's answered/addressed and I can approve.

File: .gitignore (hunk 1 above)
Comment (dreamorosi) [reply .gitignore]: Good catch, remove it in the latest commit.

**Review by ijemmy — APPROVED** [review]

**Review by flochaz — APPROVED** [review]

**Pull request merged**